app/services/platforms/youtube.ts

```diff
@@ -117,109 +183,62 @@ export class YoutubeService extends StatefulService<IYoutubeServiceState>
       );
   }
 
-  ableToStream(): Promise<void> {
-    const endpoint = 'liveBroadcasts?part=contentDetails&mine=true&broadcastType=persistent';
-    const request = new Request(`${this.apiBase}/${endpoint}&access_token=${this.oauthToken}`);
-
-    return fetch(request)
-      .then(resp => {
-        return resp.status === 403 ? this.handleForbidden(resp) : handlePlatformResponse(resp);
-      })
-      .then(() => this.SET_ENABLED_STATUS(true));
-  }
-
   fetchUserInfo() {
     return Promise.resolve({});
   }
 
-  fetchBoundStreamId(): Promise<string> {
-    const endpoint = 'liveBroadcasts?part=contentDetails&mine=true&broadcastType=persistent';
-    return platformAuthorizedRequest(
+  async fetchStreamKey(): Promise<string> {
+    const endpoint = `liveStreams?part=cdn,snippet&default=true`;
+    return platformAuthorizedRequest<IYoutubeCollection<IYoutubeLiveStream>>(
       `${this.apiBase}/${endpoint}&access_token=${this.oauthToken}`,
-    ).then(json => json.items[0].contentDetails.boundStreamId);
-  }
-
-  handleForbidden(response: Response): void {
-    response.json().then((json: any) => {
-      if (
-        json.error &&
-        json.error.errors &&
-        json.error.errors[0].reason === 'liveStreamingNotEnabled'
-      ) {
-        this.SET_ENABLED_STATUS(false);
-      }
-    });
-  }
-
-  fetchStreamKeyForId(streamId: string): Promise<string> {
-    const endpoint = `liveStreams?part=cdn&id=${streamId}`;
-    return platformAuthorizedRequest(
-      `${this.apiBase}/${endpoint}&access_token=${this.oauthToken}`,
-    ).then(json => json.items[0].cdn.ingestionInfo.streamName);
-  }
-
-  fetchStreamKey(): Promise<string> {
-    return this.fetchBoundStreamId()
-      .then(boundStreamId => this.fetchStreamKeyForId(boundStreamId))
-      .catch(() => $t('Please enable account for streaming to recieve a stream key'));
-  }
-
-  fetchChannelInfo(): Promise<IChannelInfo> {
-    return Promise.resolve(this.state.channelInfo);
-  }
-
-  getLiveStreamId(forceGet: boolean): Promise<void> {
-    if (this.state.liveStreamId && !forceGet) return Promise.resolve();
-
-    const endpoint = 'liveBroadcasts?part=id&broadcastStatus=active&broadcastType=persistent';
-
-    return platformRequest(`${this.apiBase}/${endpoint}&access_token=${this.oauthToken}`).then(
-      json => {
-        if (json.items.length) {
-          this.SET_STREAM_ID(json.items[0].id);
-        }
-      },
-    );
+    )
+      .then(streams => streams.items[0].cdn.ingestionInfo.streamName)
+      .catch(e => $t('Please enable account for streaming to recieve a stream key'));
```

Comment:
what if they error out for other reasons?

Reply:
The next request after `fetchStreamKey` will be `repopulateStreamInfo`. We handle errors better there